src/addons/par/icc/tools/icc_common.sh

```diff
@@ -3,6 +3,24 @@
 # methodology, adding/modifying options, and generating scripts.
 # Use with 'source'.
 
+# TODO: pull these into a generic shell utils or something for re-use.
+# See src/addons/synthesis/dc/tools/run-synthesis
+
+# Helper function to join arrays.
+# Example: join_by , "${FOO[@]}" #a,b,c
+# http://stackoverflow.com/a/17841619
+function join_by { local d=$1; shift; echo -n "$1"; shift; printf "%s" "${@/#/$d}"; }
+
+# Older versions of readlink, like readlink 8.4, don't support multiple
+# arguments to readlink...
+# Usage: x=$(readlink_array my_array)
+readlink_array() {
+    local ref=$1[@]
+    local v=(${!ref})
+    local output=$(join_by " " "${v[@]}")
```

Comment:
Yes, on the BWRC servers the system readlink binary does not support multiple files, so I replaced it with a shell function.

That being said, it might perhaps be cleaner to implement by own readlink script and then just put that ahead in the path. That way we have a (slightly) simpler syntax and avoid this shell function. Thoughts?